PopPUNK/models.py

```diff
@@ -1006,34 +1027,50 @@ class LineageFit(ClusterFit):
             The ranks used in the fit
     '''
 
-    def __init__(self, outPrefix, ranks, use_gpu = False):
+    def __init__(self, outPrefix, ranks, max_search_depth, reciprocal_only, count_unique_distances, dist_col = None, use_gpu = False):
         ClusterFit.__init__(self, outPrefix)
         self.type = 'lineage'
         self.preprocess = False
+        self.max_search_depth = max_search_depth
+        self.nn_dists = None # stores the unprocessed kNN at the maximum search depth
         self.ranks = []
         for rank in sorted(ranks):
             if (rank < 1):
                 sys.stderr.write("Rank must be at least 1")
                 sys.exit(0)
             else:
                 self.ranks.append(int(rank))
+        self.lower_rank_dists = {}
+        self.reciprocal_only = reciprocal_only
+        self.count_unique_distances = count_unique_distances
+        self.dist_col = dist_col
         self.use_gpu = use_gpu
 
-    def __save_sparse__(self, data, row, col, rank, n_samples, dtype):
+    def __save_sparse__(self, data, row, col, rank, n_samples, dtype, is_nn_dist = False):
       '''Save a sparse matrix in coo format
       '''
       if self.use_gpu:
           data = cp.array(data)
           data[data < epsilon] = epsilon
-          self.nn_dists[rank] = cupyx.scipy.sparse.coo_matrix((data, (cp.array(row), cp.array(col))),
-                                              shape=(n_samples, n_samples),
-                                              dtype = dtype)
+          if is_nn_dist:
+            self.nn_dists = cupyx.scipy.sparse.coo_matrix((data, (cp.array(row), cp.array(col))),
```

Comment:
These indents seem wrong

Reply:
Still using two spaces from C++ code, fixed in 2a53e80